### DISCUSSION TITLE
Definition of software

### DISCUSSION THREAD

**Discussion by wanneut** [nfdi4plants/ARC-specfication#35] [Comments and Suggestions]
The part: "Therefore, necessary software MUST be available in Workflows." needs a proper definition of "software". If you use a tool written in C#: Do we need a full copy of Windows+.NET-Runtime? Or only the source code with an compiler? Or is it OK to add some  Startup script and skip all background services that are needed? Mabe we can copy that from some freeware.

**Comment by mephenor** [comment]
I agree that this needs to be formulated a bit differently, maybe "available and runnable" or something in that direction.
Another potential issue I see here is if everything that makes runnable is to be included, we might get into some problems if parts of the software are not freely distributable.

**Comment by chgarth** [comment]
This is a somewhat tricky issue. One could define "runnable" as "executable by `cwltool` version X on a freshly installed Ubuntu` or similar, but this appear overreaching to me. "Runnable" may also refer to external resources such as Docker images etc., which we probably don't want to include in each an every ARC.

A possible path forward, as discussed in DataPLANT earlier, would be to leave the formulation as is to express intent, and tackle the reproducibility issue via ARC quality control mechanisms.
